### DISCUSSION TITLE
Building from Source on Ubuntu 20.04

### DISCUSSION THREAD

**Discussion by NuclearPowerNerd** [Snaipe/Criterion#459] [Q&A]
Hello

I'd like to give unit testing with Criterion a try. I'm attempting to build from source. I have followed the instructions [here ](https://criterion.readthedocs.io/en/latest/setup.html#installation) up to the step "Install the library to your library search path".

I am unsure how to do this. I know I can update my search path using `LD_LIBRARY_PATH` but where or to what am I to point it to? I don't see any `.so` files in any subdirectory of Criterion. Does that mean it built incorrectly? Or am I missing something?

**Comment by Snaipe** [comment]
It mostly depends on where you end up installing criterion -- so, assuming the default install prefix of /usr/local, after running `sudo ninja install`, you should end up with /usr/local/lib/libcriterion.so on your system. You can then either export LD_LIBRARY_PATH to include /usr/local/lib (and persist that export somewhere in your shell rc file), or add /usr/local/lib to /etc/ld.so.conf so that ld.so searches it by default.
